src/statistics.jl

```diff
@@ -2144,9 +2144,82 @@ function apply_statistic(stat::DodgeStatistic,
     end
 end
 
+struct QuantileBarsStatistic <: Gadfly.StatisticElement
+    quantiles::Vector{Float64}
+    # We cannot avoid these by combining our statistic with Stat.density,
+    # because we need the raw data as well as the kernel density.
+    n::Int # Number of points sampled.
+    bw::Real # Bandwidth used for the kernel density estimation.
+end
+QuantileBarsStatistic(; quantiles=[0.025, 0.975], n=256, bandwidth=-Inf) =
+    QuantileBarsStatistic(quantiles, n, bandwidth)
+
+input_aesthetics(stat::QuantileBarsStatistic) = [:x]
+output_aesthetics(stat::QuantileBarsStatistic) = [:x, :y, :xend, :yend]
+
+"""
+    Stat.quantile_bars[(; quantiles=[0.025, 0.975], bar_width=0.1, n=256, bandwidth=-Inf)]
 
+Transform the point in $(aes2str(input_aesthetics(quantile_bars()))) into a set of 
+$(aes2str(output_aesthetics(quantile_bars()))) points. These points can then be drawn
+via [`Geom.segment`](@ref Gadfly.Geom.segment).
+"""
+const quantile_bars = QuantileBarsStatistic
+
+"""
+    _calculate_quantile_bar(stat::QuantileBarsStatistic, aes)
 
+Helper function for `apply_statistic(stat::QuantileBarsStatistic, ...)`.
+"""
+function _calculate_quantile_bar(stat::QuantileBarsStatistic, xs)
+    isa(xs[1], Real) || error("Kernel density estimation only works on Real types.")
+
+    window = stat.bw <= 0.0 ? KernelDensity.default_bandwidth(xs) : stat.bw
+    k = KernelDensity.kde(xs, bandwidth=window, npoints=stat.n)
+
+    x = quantile(xs, stat.quantiles)
+    y = fill(0, length(xs))
```

Comment:
Change to `y = zeros(length(x))`. Note `length(x)` not `length(xs)` (ideally y should be the same length as x).